android/lib/models/post_model.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:android/models/user_model.dart';
+import 'package:android/pages/event_page.dart';
+import 'package:flutter/material.dart';
+import 'package:android/models/models.dart';
+
+class Post {
+  final String type;
+  final int id;
+  final User creator;
+  final String title;
+  final String description;
+  final String? imageUrl;
+
+  Post({
+    required this.type,
+    required this.id,
+    required this.creator,
+    required this.title,
+    required this.description,
+    required this.imageUrl,
+  });
+
+  Widget pageRoute() {
+    return EventPage(id: id);
```

Comment:
I am not sure if this Post page is ever going to redirect to an Art Item Page (or actually even render them).